packages/eslint-plugin/src/rules/await-thenable.ts

```diff
@@ -62,6 +72,44 @@ export default createRule({
           });
         }
       },
+
+      ForOfStatement(node): void {
```

Comment:
```suggestion
      'ForOfStatement[await=true]'(node): void {
```

And you can remove the condition below